sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -713,6 +713,7 @@ case class MapConcat(children: Seq[Expression])
     }
   }
 
+  override def stateful: Boolean = true
```

Comment:
We should add this to `MapFromEntries`, `CreateMap`, and `CreateMap`, too?

Reply:
good catch!